content/docs/security/csrf.md

```diff
@@ -0,0 +1,201 @@
+
+> Cross-Site Request Forgery (CSRF) is an attack that forces an end user to execute unwanted actions on a web application in which they're currently authenticated. With a little help of social engineering (such as sending a link via email or chat), an attacker may trick the users of a web application into executing actions of the attacker's choosing. If the victim is a normal user, a successful CSRF attack can force the user to perform state changing requests like transferring funds, changing their email address, and so forth. If the victim is an administrative account, CSRF can compromise the entire web application.
+
+To help prevent CSRF attacks, you can enable the CSRF filter on your gateway or a specific route. For each route that you apply the CSRF policy to, the filter checks to make sure that a request's origin matches its destination. If the origin and destination do not match, a 403 Forbidden error code is returned. 
+
+{{% callout type="info" %}}
+Note that because CSRF attacks specifically target state-changing requests, the filter only acts on HTTP requests that have a state-changing method such as `POST` or `PUT`.
+{{% /callout %}}
+
+{{% callout type="info"  %}}
+To learn more about CSRF, you can try out the [CSRF sandbox](https://www.envoyproxy.io/docs/envoy/latest/start/sandboxes/csrf) in Envoy. 
+{{% /callout%}}
+
+## Before you begin
+
+{{< reuse "docs/snippets/prereq.md" >}}
+
+## Set up CSRF 
+
+Use a TrafficPolicy resource to define your CSRF rules. 
+
+1. Create a TrafficPolicy resource to define your CSRF rules. The following example allows request from only the `allowThisOne.example.com` origin.
+   
+   ```yaml
+   kubectl apply -f- <<EOF
+   apiVersion: gateway.kgateway.dev/v1alpha1
+   kind: TrafficPolicy
+   metadata:
+     name: csrf
+     namespace: httpbin
+   spec:
+     targetRefs:
+     - group: gateway.networking.k8s.io
+       kind: HTTPRoute
+       name: httpbin-csrf
+     csrf:
+       percentageEnabled: 100
+       additionalOrigins:
+       - exact: allowThisOne.example.com
+         ignoreCase: false
+   EOF
+   ```
+
+   {{< reuse "docs/snippets/review-table.md" >}} For more information, see the [API docs](../../reference/api/#csrfpolicy).
+
+   | Field | Description |
+   |-------|-------------|
+   | `percentageEnabled` | The percentage of requests for which the CSRF policy is enabled. A value of `100` means that all requests are enforced by the CSRF rules. |
+   | `additionalOrigins` | Additional origins that the CSRF policy allows, besides the destination origin. Possible values include `exact`, `prefix`, `suffix`, `contains`, `safeRegex`, and an `ignoreCase` boolean. At least one of `exact`, `prefix`, `suffix`, `contains`, or `safeRegex` must be set. The example allows requests from the `allowThisOne.example.com` origin. |
+
+2. Create an HTTPRoute resource for the httpbin app that applies the TrafficPolicy resource that you just created. 
```

Comment:
we add the policy via extensionref, but you actually already have a targetRef in the trafficpolicy. we would need to remove one or the other